src/core/tests/type_prop/stft.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/stft.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "common_test_utils/type_prop.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/subtract.hpp"
+
+namespace ov {
+namespace test {
+
+using op::v0::Constant;
+using op::v0::Parameter;
+using testing::HasSubstr;
+
+class TypePropSTFTTest : public TypePropOpTest<op::v15::STFT> {
+public:
+    bool transform_frames = true;
+};
+
+TEST_F(TypePropSTFTTest, default_ctor) {
+    const auto op = make_op();
+    const auto signal = std::make_shared<Parameter>(element::f32, PartialShape{4, 48});
+    const auto window = std::make_shared<Parameter>(element::f32, PartialShape{7});
+    const auto frame_size = Constant::create<int32_t>(element::i32, {}, {11});
+    const auto frame_step = Constant::create<int32_t>(element::i32, {}, {3});
+
+    op->set_arguments(OutputVector{signal, window, frame_size, frame_step});
+    op->set_transpose_frames(true);
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_output_size(), 1);
+    EXPECT_EQ(op->get_input_size(), 4);
+    EXPECT_EQ(op->get_output_element_type(0), element::f32);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{4, 6, 13, 2}));
+}
+
+TEST_F(TypePropSTFTTest, all_inputs_as_params_static_shapes) {
+    const auto signal = std::make_shared<Parameter>(element::f32, PartialShape{4, 48});
+    const auto window = std::make_shared<Parameter>(element::f32, PartialShape{7});
```

Comment:
Use others floating-point precision for tests

Reply:
Added test with f16